File: check_state.js
const { database } = require('./firebase.config.js'); // Assuming you have a CJS version or we can just fetch via REST

const https = require('https');

function fetchFirebase(path) {
    return new Promise((resolve, reject) => {
        https.get(`https://ujjwal-7a8c8-default-rtdb.firebaseio.com/${path}.json`, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => resolve(JSON.parse(data)));
        }).on('error', reject);
    });
}

async function check() {
    const bins = await fetchFirebase('bins');
    if (!bins) {
        console.log("NO BINS IN DB!");
        return;
    }

    const binValues = Object.values(bins);
    console.log(`Total Bins in DB: ${binValues.length}`);

    const kandigaiBins = binValues.filter(b => b.areaName && b.areaName.toLowerCase() === 'kandigai');
    console.log(`Total Kandigai Bins: ${kandigaiBins.length}`);
    if (kandigaiBins.length > 0) {
        console.log("Sample Kandigai Bin:");
        console.log(kandigaiBins[0]);
    }

    // Also check users
    const users = await fetchFirebase('users');
    console.log(`Total Users: ${Object.values(users || {}).length}`);
    const admins = Object.values(users || {}).filter(u => u.role === 'admin' || u.role === 'superadmin');
    console.log("Admins:");
    admins.forEach(a => {
        console.log(`- ${a.email} | Location: ${a.location}`);
    });
}

check();

File: check_state.py
import urllib.request
import json

BASE_URL = "https://ujjwal-7a8c8-default-rtdb.firebaseio.com"

def fetch(path):
    url = f"{BASE_URL}/{path}.json"
    req = urllib.request.Request(url)
    try:
        with urllib.request.urlopen(req) as res:
            return json.loads(res.read().decode())
    except Exception as e:
        print(f"Error fetching {path}: {e}")
        return {}

bins = fetch("bins")
if not bins:
    print("NO BINS IN DB!")
else:
    bin_values = list(bins.values())
    print(f"Total Bins in DB: {len(bin_values)}")
    kandigai_bins = [b for b in bin_values if b and b.get("areaName", "").lower() == "kandigai"]
    print(f"Total Kandigai Bins: {len(kandigai_bins)}")
    if kandigai_bins:
        print("Sample Kandigai Bin:", kandigai_bins[0])

users = fetch("users")
if users:
    admins = [u for u in users.values() if u and u.get("role") in ["admin", "superadmin"]]
    print("Admins:")
    for a in admins:
        print(f"- {a.get('email')} | Location: {a.get('location')} | Area: {a.get('area')}")
